google/cloud/bigquery/schema.py

```diff
@@ -921,3 +915,83 @@ def from_api_repr(cls, resource: dict) -> SerDeInfo:
         config = cls("")
         config._properties = copy.deepcopy(resource)
         return config
+
+
+class Schema:
+    # TODO docstrings and type hints
+    def __init__(self, fields=None, foreign_type_info=None):
```

Comment:
Presumably we have these `fields` and `foreign_type_info` arguments because we expect users to be constructing `Schema` objects directly? If so, accepting an Iterable of `SchemaField` or `dict` would be the most user-friendly IMO.

Reply:
Resolving. I missed that `_fields` was a property with a setter. I think your code does actually convert Iterable of SchemaField or dict to the correct API representation in _properties